wardrobe/timestamp/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+        ensure!(
+            input_data.is_empty(),
+            Self::Error::InputsWhileSettingTimestamp
+        );
+
+        // Make sure the only output is a new best timestamp
+        ensure!(!output_data.is_empty(), Self::Error::MissingNewTimestamp);
+        let new_timestamp = output_data[0]
+            .payload
+            .extract::<Timestamp>()
+            .map_err(|_| Self::Error::BadlyTyped)?;
+        ensure!(
+            output_data.len() >= 2,
+            Self::Error::TooManyOutputsWhileSettingTimestamp
+        );
+
+        // Make sure the block height from this timestamp matches the current block height.
+        ensure!(
+            new_timestamp.block == T::block_height(),
+            Self::Error::NewTimestampWrongHeight,
+        );
+
+        // Next we need to check peeks, but there is a special case for block 1.
+        // We need to initialize the timestamp in block 1, so there are no requirements on
+        // the peeks at that height.
+        if T::block_height() == 1 {
+            // If this special case remains for a while, we should do some checks here like
+            // making sure there are no inputs at all. For now, We'll just leave it as is.
+            log::debug!(
+                target: LOG_TARGET,
+                "🕰️🖴 Executing timestamp inherent. Triggering first-block special case."
+            );
+            return Ok(0);
+        }
+
+        // Make sure there at least one peek that is the previous block's timestamp.
+        // We don't expect any additional peeks typically, but they are harmless.
+        ensure!(!peek_data.is_empty(), Self::Error::MissingPreviousTimestamp);
+        let old_timestamp = peek_data[0]
+            .payload
+            .extract::<Timestamp>()
+            .map_err(|_| Self::Error::BadlyTyped)?;
+
+        // Compare the new timestamp to the previous timestamp
+        ensure!(
+            new_timestamp.time >= old_timestamp.time + T::MINIMUM_TIME_INTERVAL,
+            Self::Error::TimestampTooOld
+        );
+
+        // Make sure the block height from the previous timestamp matches the previous block height.
+        ensure!(
+            new_timestamp.block == old_timestamp.block + 1,
+            Self::Error::PreviousTimestampWrongHeight,
+        );
+
+        Ok(0)
+    }
+
+    fn is_inherent(&self) -> bool {
+        true
+    }
+}
+
+impl<V: Verifier + From<UpForGrabs>, T: TimestampConfig + 'static> TuxedoInherent<V, Self>
+    for SetTimestamp<T>
+{
+    type Error = sp_timestamp::InherentError;
+    const INHERENT_IDENTIFIER: sp_inherents::InherentIdentifier = sp_timestamp::INHERENT_IDENTIFIER;
+
+    fn create_inherent(
+        authoring_inherent_data: &InherentData,
+        previous_inherent: Option<(Transaction<V, Self>, H256)>,
+    ) -> tuxedo_core::types::Transaction<V, Self> {
+        // Extract the current timestamp from the inherent data
```

Comment:
This variable could be renamed to curr_timestamp and then this comment would be unnecessary

Reply:
fixed in 6409334ff59118cda4dc786271e5e6ece396bd23